coralproject/talk
# Convert Coral into a mono-repo with 3 main projects

## What does this PR do?

Breaks Coral up into a mono-repo along the lines of `client`, `common`, and `server`. Who are analogous to the imports you are likely familiar with such as `coral-server/...`, `coral-client/...` and `coral-common/...`.

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?

No changes to schema.

## Does this PR introduce any new environment variables or feature flags?

No new env vars.

## If any indexes were added, were they added to `INDEXES.md`?

No new indices.

## How do I test this PR?

Testing watch:
- run `sh initialize.sh`
- `cd` into `client/` and `npm run watch`
- open another terminal and `cd` into `server/` and `npm run watch`
- visit `http://localhost:8080` and make sure the stream and other utilities work

Testing builds:
- `cd` into `client/` and `npm run build`
- open another terminal and `cd` into `server/` and `npm run build`

Testing a docker build:

- For `arm` based macs:
  - use: `docker buildx build --platform=linux/amd64 -t gcr.io/the-coral-project/coral:test-0.1.0 -f Dockerfile .`
- For `Intel, x86/64` macs:
  - use: `docker build -t gcr.io/the-coral-project/coral:test-0.1.0 -f Dockerfile .`

## Where any tests migrated to React Testing Library?

No

## How do we deploy this PR?
